### DISCUSSION TITLE
Error while running directus.

### DISCUSSION THREAD

**Discussion by bhagya1101** [directus/directus#16933] [Archive]
Hi,
When I am running Directus on browser, I am getting : {
  "errors": [
    {
      "message": "Connection lost: The server closed the connection.",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}

I am using MySQL database and I don't know what is this error and how to resolve it??
Any help will be appreciated.
Thanks.

**Comment by br41nslug** [comment]
Any errors on the Directus side when accessing it in the browser?

**Reply by bhagya1101** [reply]
![Screenshot 2022-12-27 at 2 01 00 PM](https://user-images.githubusercontent.com/117433805/209637324-312fb0bc-1541-41ff-9801-905b75ff6d19.png)

This is the error I get on my terminal
and this is on directus,

![Screenshot 2022-12-27 at 2 02 13 PM](https://user-images.githubusercontent.com/117433805/209637510-59492c2d-7fcc-418a-bd68-335bc2efa79e.png)

**Reply by br41nslug** [reply]
Looks like Directus is unable to connect/authentication to the database. Are the connection details correct and tested? (can you login to the database using a local database management tool?)

**Reply by bhagya1101** [reply]
Yes, connection details are correct and I am able to access database locally in my machine perfectly with mysql.

Every time, when this error is displayed, after refreshing browser for average of 8 times, Directus loads perfectly and if refreshed again, the error pops up again .

**Reply by br41nslug** [reply]
Are you getting any database errors like max connections reached or similar?

**Reply by bhagya1101** [reply]
Nope, no such errors at all

**Reply by bhagya1101** [reply]
I am currently running it on localhost

**Reply by bhagya1101** [reply]
This is the error I am getting : 
`Request failed with status code 500`

**Comment by bhagya1101** [comment]
I had a question in my mind that if I want to run a custom module in which I have made changes, how do I run it. 
Normally, I do `npm run build ` and then the I copy the index.js dist file to `extensions/modules/data/index.js `
and then `npm start`...

Every time I make changes in module.vue file, I follow the above process.

Is it the right process or is there some other way around ??

**Reply by alexvdvalk** [reply]
You'll need to setup your npm run build to output the index.js to the correct folder. There is an environmental variable called EXTENSIONS_AUTO_RELOAD which auto reloads the server when any extensions are modified.

https://docs.directus.io/self-hosted/config-options.html

**Reply by br41nslug** [reply]
As Alex mentioned the `EXTENSIONS_AUTO_RELOAD` will reload extensions when changed.
Combine that with:
- Setting the build output destination directly to your extensions folder
  1. For docker you can mount the `dist/` folder directly 
    ```
      volumes:
        - "./dist:/directus/extensions/hooks/directus-extension-custom-hook"
     ```
   2. For npm installations set the `"path"` in your `package.json`
     ```
     "directus:extension": {
		"type": "hook",
		"path": "[PATH/TO/YOUR/EXTENSION/FOLDER]/index.js",
		"source": "src/index.ts",
		"host": "^9.22.1"
     },
     ```
-  and running `npm run dev` (which should watch your local source files)

That should get you auto-reloading all the way 😄

**Reply by bhagya1101** [reply]
Thanks a lot both of you for helping...

**Comment by bhagya1101** [comment]
> Hi, When I am running Directus on browser, I am getting : { "errors": [ { "message": "Connection lost: The server closed the connection.", "extensions": { "code": "INTERNAL_SERVER_ERROR" } } ] }
> 
> I am using MySQL database and I don't know what is this error and how to resolve it?? Any help will be appreciated. Thanks.

Please help me about this. I do not know why I am getting this error. Every time I have to reload it for 8 times on average. 
Currently I am working on project, I hope that you will help me for solving this error

**Comment by lisaiceland** [comment]
Hi ;-) 
This has NOT been resolved. 
Is there a default environment variable we can setup to solve this?

**Reply by br41nslug** [reply]
This is a 1.5+ year old archived and obsolete discussion. If you're running into an issue have a look at [issues](https://github.com/directus/directus/issues) for whether it already exists or create a new one.
